### ISSUE TITLE
SlideMenu: Missing keyboard navigation

### ISSUE THREAD

**Issue by Teutron-flo** [primefaces/primefaces#14418]
### Describe the bug

Right now there is no possibility to focus the Slide Menu component or navigate through the menu entries only with the keyboard. The reason for the same is that all elements are basicly rendered with tabindex -1 and there is no function to change it during user interaction.

The current state can be tested on the showcase page -> https://showcase.primefaces.org/ui/menu/slideMenu.xhtml

To solve the problem the following things have to be done:

1. All menu entries needs a intial value on the first rendering:

Main menu entries        -> tabindex 0
Sub menu links              -> tabindex 0
Sub menu entries          -> tabindex -1
Sub menu back button -> tabindex -1

2. Add a function to switch tabindex when you move to a sub menu and also when you go back to main menu

3. Add js functions for hover and focus to add the necessary styleclasses

4. Make sub menu back button "clickable"

5. Add an addtional fix for overlay slide menus.

### PrimeFaces edition

None

### PrimeFaces version

15.0.10

### JSF implementation

All

### JSF version

4.0

### Java version

21

**Comment by Teutron-Florian** [comment]
I added a first PR to address points 1 - 4. But I'm right now not able to fix it for overlay option. 
The issue here is that the menu is added to the bottom of the html page, thats why you can not directly navigate from the menu button to the overlay menu.
Maybe somebody has an hint or a best practice template for the same...

**Comment by melloware** [comment]
@Teutron-flo for the OverlayMenu i think you have to on `show` find the first menu item and call `trigger("focus")` to then start the keyboard navigation.

**Comment by melloware** [comment]
@Teutron-flo  add this `PrimeFaces.queueTask` line near 248

```js
    override show(): void {
		const $this = this;
        if (this.transition) {
            this.transition.show({
                onEnter: () => {
                    if (!this.rendered) {
                        this.render();
                    }
                    PrimeFaces.nextZindex(this.jq);
                    this.align();
                },
                onEntered: () => {
                    this.bindPanelEvents();
					PrimeFaces.queueTask(function() {$this.content.find('a.ui-menuitem-link:first').trigger('focus')}, 0);
                }
            });
        }
    }
```

**Comment by Teutron-Florian** [comment]
@melloware Thanks for sharing the solution here :) . I added it to the code and made also the addtional change that you had requested.

**Comment by melloware** [comment]
Thanks for the PR!  This will be in 16.0.0
